tests/rptest/tests/schema_registry_test.py

```diff
@@ -1089,6 +1089,34 @@ def test_post_subjects_subject_versions_version_many(self):
             assert result_raw.status_code == requests.codes.ok
             assert result_raw.json()["id"] == 1
 
+    @cluster(num_nodes=3)
+    def test_post_subjects_subject_versions_metadata_ruleset(self):
+        """
+        Verify posting a schema with metatada and ruleSet
+        These are not supported, but if they're null, we let it pass.
+        """
+
+        topic = create_topic_names(1)[0]
+
+        self.logger.debug("Dump the schema with null metadata and ruleSet")
+        schema_1_data = json.dumps({
+            "schema": schema1_def,
+            "metadata": None,
+            "ruleSet": None
```

Comment:
nitpick: should we assert that we reject requests where these are non-null?

Reply:
> nitpick: should we assert that we reject requests where these are non-null?

Probably. I propose a new PR for that - improvement over perfection and all that. We could also check that the check-compatible endpoint is also happy with it, but currently the same code is used, so I presume so.

Reply:
>no reason not to handle empty objects

yeah, i was thinking more along those line. the test as formulated doesn't tell much about what we _do_ do with other stuff. follow-on sgtm.